google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.java

```diff
@@ -137,17 +140,28 @@ public ApiFuture<AppendRowsResponse> append(JSONArray jsonArr, long offset)
 
       ProtoRows.Builder rowsBuilder = ProtoRows.newBuilder();
       // Any error in convertJsonToProtoMessage will throw an
-      // IllegalArgumentException/IllegalStateException/NullPointerException and will halt
-      // processing
-      // of JSON data.
+      // IllegalArgumentException/IllegalStateException/NullPointerException which will be collected
+      // into a Map of roe indexes to error messages. After the coverstion is finished an
+      // AppendSerializtionError exception that contains all the conversion errors will be thrown.
       long currentRequestSize = 0;
+      Map<Integer, String> rowIndexToErrorMessage = new HashMap<>();
       for (int i = 0; i < jsonArr.length(); i++) {
         JSONObject json = jsonArr.getJSONObject(i);
-        Message protoMessage =
-            JsonToProtoMessage.convertJsonToProtoMessage(
-                this.descriptor, this.tableSchema, json, ignoreUnknownFields);
-        rowsBuilder.addSerializedRows(protoMessage.toByteString());
-        currentRequestSize += protoMessage.getSerializedSize();
+        try {
+          Message protoMessage =
+              JsonToProtoMessage.convertJsonToProtoMessage(
+                  this.descriptor, this.tableSchema, json, ignoreUnknownFields);
+          rowsBuilder.addSerializedRows(protoMessage.toByteString());
+          currentRequestSize += protoMessage.getSerializedSize();
+        } catch (IllegalArgumentException
+            | IllegalStateException
+            | NullPointerException exception) {
```

Comment:
Do we want to catch all of these?  IllegalState is ambiguous, but NullPointerException seems like a more serious error indicating something is wrong, rather than a "row-level" error

Reply:
In [JsonToProtoMessage#convertJsonToProtoMessage](https://github.com/googleapis/java-bigquerystorage/blob/e85522adaff0f61780647d40e2ba3a654cd403a0/google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessage.java#L156 ) there are some Null and state checks, some of them are not relevant to a single line and some are, for example ` Preconditions.checkState(json.length() != 0, "JSONObject is empty.");` is relevant for a single row and should be reported as such.

I'm not sure how we can even get to a point from JsonStreamWriter that we will have null values in JsonObjects, so I will remove the catch for null and state exceptions for now, and add them later if we see it's needed